src/ice_shelf/MOM_ice_shelf_diag_mediator.F90

```diff
@@ -0,0 +1,812 @@
+  real, pointer, dimension(:,:)   :: mask2d_comp => null() !< A 2-d mask on the computational domain for this diagnostic
+  real, pointer, dimension(:,:,:) :: mask3d => null()      !< A 3-d mask for this diagnostic
+end type diag_type
+
+!>   The SIS_diag_ctrl data type contains times to regulate diagnostics along with masks and
+!! axes to use with diagnostics, and a list of structures with data about each diagnostic.
+type, public :: diag_ctrl
+  integer :: doc_unit = -1 !< The unit number of a diagnostic documentation file.
+                           !! This file is open if doc_unit is > 0.
+
+  ! The following fields are used for the output of the data.
+  ! These give the computational-domain sizes, and are relative to a start value
+  ! of 1 in memory for the tracer-point arrays.
+  integer :: is  !< The start i-index of cell centers within the computational domain
+  integer :: ie  !< The end i-index of cell centers within the computational domain
+  integer :: js  !< The start j-index of cell centers within the computational domain
+  integer :: je  !< The end j-index of cell centers within the computational domain
+  ! These give the memory-domain sizes, and can be start at any value on each PE.
+  integer :: isd !< The start i-index of cell centers within the data domain
+  integer :: ied !< The end i-index of cell centers within the data domain
+  integer :: jsd !< The start j-index of cell centers within the data domain
+  integer :: jed !< The end j-index of cell centers within the data domain
+  real :: time_int              !< The time interval in s for any fields that are offered for averaging.
+  type(time_type) :: time_end   !< The end time of the valid interval for any offered field.
+  logical :: ave_enabled = .false. !< .true. if averaging is enabled.
+
+  !>@{ The following are 3D and 2D axis groups defined for output.  The names indicate
+  !! the horizontal locations (B, T, Cu, or Cv), vertical locations (L, i, or 1) and
+  !! thickness categories (c, c0, or 1).
+  type(axesType) :: axesBL, axesTL, axesCuL, axesCvL
+  type(axesType) :: axesBi, axesTi, axesCui, axesCvi
+  type(axesType) :: axesBc, axesTc, axesCuc, axesCvc
+  type(axesType) :: axesBc0, axesTc0, axesCuc0, axesCvc0
+  type(axesType) :: axesB1, axesT1, axesCu1, axesCv1
+  !!@}
+  type(axesType) :: axesZi !< A 1-D z-space axis at interfaces
+  type(axesType) :: axesZL !< A 1-D z-space axis at layer centers
+
+  ! Mask arrays for diagnostics
+  real, dimension(:,:),   pointer :: mask2dT   => null() !< 2D mask array for cell-center points
+  real, dimension(:,:),   pointer :: mask2dBu  => null() !< 2D mask array for cell-corners
+  real, dimension(:,:),   pointer :: mask2dCu  => null() !< 2D mask array for east-faces
+  real, dimension(:,:),   pointer :: mask2dCv  => null() !< 2D mask array for north-faces
+  real, dimension(:,:,:), pointer :: mask3dTL  => null() !< 3D mask array for layer cell-centers
```

Comment:
There are no vertical axes used or defined for the ice-shelf diag mediator, so why are all of these 3d mask variables needed?  It seems to me that the whole point here is to have a 2-d diag mediator type for the use of the ice-shelf code.

Also, it seems like a recipe for confusion to use the same type and subroutine names for the ice-shelf diag mediator as are used by the broader MOM6 diag mediator. Would it make sense to give these types and subroutines unique names to avoid confusion?

Reply:
Removed the vertical axis and mask lines.

I don't see a problem with namespace conflict, i.e. we are not simultaneously dealing with MOM6 and IceShelf diagnostics within a single module. It would perhaps make sense to *not* have separate diag mediators for  MOM6, SIS2 and the IceShelf. They could be combined in a single module which is overloaded to work with all 3 data structures.